### PR TITLE
Unit tests for Facilitator.start and Facilitator.stop

### PR DESCRIPTION
**Pull request by abhayks1**
PR contains below:

- Unit test for Facilitator.start
- Unit test for Facilitator.stop
- Removed unused code

### CODE REVIEW

File: test/Facilitator/stop.test.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import * as sinon from 'sinon';
+
+import Facilitator from '../../src/Facilitator';
+import GraphClient from '../../src/GraphClient'
+import Subscriber from '../../src/Subscriber'
+import {Config} from "../../src/Config";
```
Comment (sarvesh-ost) [inline test/Facilitator/stop.test.ts]:
```suggestion
import {Config} from '../../src/Config';
```

File: test/Facilitator/stop.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import SpyAssert from "../utils/SpyAssert";
```
Comment (sarvesh-ost) [inline test/Facilitator/stop.test.ts]:
```suggestion
import SpyAssert from '../utils/SpyAssert';
```

File: test/Facilitator/stop.test.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+describe('Facilitator.stop()', () => {
+  it('stop should work properly', async () => {
+    const mockGraphClient = sinon.createStubInstance(GraphClient);
+    const graphClientSpy = sinon.replace(
+      GraphClient,
+      'getClient',
+      sinon.fake.returns(mockGraphClient),
+    );
+    const subscriberStub = sinon.stub(Subscriber.prototype, "subscribe");
```
Comment (sarvesh-ost) [inline test/Facilitator/stop.test.ts]:
```suggestion
    const subscriberStub = sinon.stub(Subscriber.prototype, 'subscribe');
```

File: test/Facilitator/stop.test.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    const subGraphDetails = Facilitator.getSubscriptionDetails();
+    const configStub = sinon.createStubInstance(Config);
+    const facilitator = new Facilitator(configStub);
+    await facilitator.start();
+
+    SpyAssert.assert(
```
Comment (sarvesh-ost) [inline test/Facilitator/stop.test.ts]: Duplicate, this is already asserted in `start` test.

**Comment by abhayks1** [comment]
Available for review again 🚀 🎸

File: test/Facilitator/start.test.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import * as sinon from 'sinon';
+
+import Facilitator from '../../src/Facilitator';
+import GraphClient from '../../src/GraphClient';
+import Subscriber from '../../src/Subscriber';
+import { Config } from '../../src/Config';
+import SpyAssert from '../utils/SpyAssert';
+
+describe('Facilitator.start()', () => {
+  it('should work with correct parameters', async () => {
```
Comment (sarvesh-ost) [inline test/Facilitator/start.test.ts]: There is no parameters.

Comment (sarvesh-ost) [inline test/Facilitator/start.test.ts]:
```suggestion
  it('should start facilitation', async () => {
```

File: test/Facilitator/stop.test.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import * as sinon from 'sinon';
+
+import Facilitator from '../../src/Facilitator';
+import GraphClient from '../../src/GraphClient';
+import Subscriber from '../../src/Subscriber';
+import { Config } from '../../src/Config';
+import SpyAssert from '../utils/SpyAssert';
+
+describe('Facilitator.stop()', () => {
+  it('stop should work properly', async () => {
```
Comment (sarvesh-ost) [inline test/Facilitator/stop.test.ts]:
```suggestion
  it('should stop facilitation', async () => {
```

File: test/Facilitator/stop.test.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+      'getClient',
+      sinon.fake.returns(mockGraphClient),
+    );
+    const subscriberStub = sinon.stub(Subscriber.prototype, 'subscribe');
+
+    const configStub = sinon.createStubInstance(Config);
+    const facilitator = new Facilitator(configStub);
+    await facilitator.start();
+
+    const unSubscriberStub = sinon.stub(Subscriber.prototype, 'unsubscribe');
```
Comment (sarvesh-ost) [inline test/Facilitator/stop.test.ts]:
```suggestion
    const unSubscribeStub = sinon.stub(Subscriber.prototype, 'unsubscribe');
```

File: test/Facilitator/stop.test.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+describe('Facilitator.stop()', () => {
+  it('stop should work properly', async () => {
+    const mockGraphClient = sinon.createStubInstance(GraphClient);
+    sinon.replace(
+      GraphClient,
+      'getClient',
+      sinon.fake.returns(mockGraphClient),
+    );
+    const subscriberStub = sinon.stub(Subscriber.prototype, 'subscribe');
```
Comment (sarvesh-ost) [inline test/Facilitator/stop.test.ts]:
```suggestion
    const subscribeStub = sinon.stub(Subscriber.prototype, 'subscribe');
```

**Pull request merged**